src/test/sql/sqlite_testrunner/sqlite_testrunner.cpp

```diff
@@ -81,6 +81,14 @@ void SQLiteTestRunner::SetUp() {
 
   auto& table_cache = table_cache_iter->second;
 
+  // In case the previous SQL query was not executed successfully, we always reset all tables
+  // because we cannot be sure that the dirty flags have been properly set.
+  for (auto& [_, table_cache_entry] : table_cache) {
+    if (!_last_run_successful) {
```

Comment:
maybe move if statement outside of the loop

Reply:
Yepp, thanks